astroquery/gaia/core.py

```diff
@@ -0,0 +1,561 @@
+        dump_to_file : bool, optional, default 'False'
+            if True, the results are saved in a file instead of using memory
+        background : bool, optional, default 'False'
+            when the job is executed in asynchronous mode, this flag specifies whether 
+            the execution will wait until results are available
+        upload_resource: str, optional, default None
+            resource to be uploaded to UPLOAD_SCHEMA
+        upload_table_name: str, required if uploadResource is provided, default None
+            resource temporary table name associated to the uploaded resource
+
+        Returns
+        -------
+        A Job object
+        """
+        return self.__gaiatap.launch_job_async(query, 
+                                               name=name, 
+                                               output_file=output_file, 
+                                               output_format=output_format, 
+                                               verbose=verbose, 
+                                               dump_to_file=dump_to_file, 
+                                               background=background, 
+                                               upload_resource=upload_resource, 
+                                               upload_table_name=upload_table_name)
+    
+    def load_async_job(self, jobid=None, name=None, verbose=False):
+        """Loads an asynchronous job
+        TAP & TAP+
+        
+        Parameters
+        ----------
+        jobid : str, mandatory if no name is provided, default None
+            job identifier
+        name : str, mandatory if no jobid is provided, default None
+            job name
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        A Job object
+        """
+        return self.__gaiatap.load_async_job(jobid, name, verbose)
+    
+    def search_async_jobs(self, jobfilter=None, verbose=False):
+        """Searches for jobs applying the specified filter
+        TAP+ only
+        
+        Parameters
+        ----------
+        jobfilter : JobFilter, optional, default None
+            job filter
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of Job objects
+        """
+        return self.__gaiatap.search_async_jobs(jobfilter, verbose)
+
+    def list_async_jobs(self, verbose=False):
+        """Returns all the asynchronous jobs
+        TAP & TAP+
+        
+        Parameters
+        ----------
+        verbose : bool, optional, default 'False' 
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of Job objects
+        """
+        return self.__gaiatap.list_async_jobs(verbose)
+    
+    def __query_object(self, coordinate, radius=None, width=None, height=None, 
```

Comment:
I think the usual way we follow is to use singe "_" for private functions/properties. However it's fine to leave it as is, since it's a big module and the usage is consistent within, and the module comes with I guess long term support from your/ESA side.